builder/azure/arm/config.go

```diff
@@ -1324,6 +1354,26 @@ func assertRequiredParametersSet(c *Config, errs *packersdk.MultiError) {
 			}
 
 		}
+
+		if c.SharedGalleryDestination.SigDestinationConfidentialVMImageEncryptionType != "" {
+			if ok, err := assertAllowedSigDestinationConfidentialVMImageEncryptionType(c.SharedGalleryDestination.SigDestinationConfidentialVMImageEncryptionType, "confidential_vm_image_encryption_type"); !ok {
+				errs = packersdk.MultiErrorAppend(errs, err)
+			}
+
+			// Ensure no disk encryption set is set when not using CMK encryption
+			for _, r := range c.SharedGalleryDestination.SigDestinationTargetRegions {
+				if r.DiskEncryptionSetId != "" && c.SharedGalleryDestination.SigDestinationConfidentialVMImageEncryptionType != string(galleryimageversions.ConfidentialVMEncryptionTypeEncryptedWithCmk) {
+					errs = packersdk.MultiErrorAppend(errs, fmt.Errorf("confidential_vm_image_encryption_type must be set to \"EncryptedWithCmk\" when passing a disk_encryption_set_id in the target_region block"))
+				}
+			}
+
+			// Ensure if the encryption type is set to EncryptedWithCmk, and the encryption type is set to ConfidentialVM, that the a disk encryption id is set
+			if c.SecurityType == constants.ConfidentialVM && c.SharedGalleryDestination.SigDestinationConfidentialVMImageEncryptionType == string(galleryimageversions.ConfidentialVMEncryptionTypeEncryptedWithCmk) {
+				if c.DiskEncryptionSetId == "" {
+					errs = packersdk.MultiErrorAppend(errs, fmt.Errorf("when using a confidential vm as source to an cvm image version and a confidential_vm_image_encryption_type of \"EncryptedWithCmk\", the source cvm must have a disk_encryption_set_id set"))
```

Comment:
Nice! This is an awesome case to add and will save users a lot of time, good thinking!

Reply:
I also was able to add an else statement here, to validate the disk_encryption_ids in the target region blocks!